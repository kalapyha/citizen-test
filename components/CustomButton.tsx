import React from "react";
import { TouchableOpacity, Text, StyleSheet } from "react-native";
import Colors from "../utils/colors";

type ButtonProps = {
  title: string;
  onPress: () => void;
  backgroundColor?: string;
  textColor?: string;
  borderColor?: string;
  buttonStylesOverride?: any;
  image?: JSX.Element;
};

function CustomButton({
  title,
  onPress,
  backgroundColor,
  textColor,
  borderColor,
  buttonStylesOverride,
  image,
}: ButtonProps) {
  return (
    <TouchableOpacity
      style={[
        styles.button,
        {
          backgroundColor: backgroundColor || Colors.primary500,
          borderColor: borderColor || Colors.primary500,
          ...buttonStylesOverride,
        },
      ]}
      onPress={onPress}
    >
      <Text style={[styles.buttonText, { color: textColor || Colors.white }]}>
        {title}
      </Text>
      <Text style={{ textAlign: "center", marginBottom: -5, marginLeft: 5 }}>
        {image}
      </Text>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  button: {
    display: "flex",
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center",
    borderRadius: 5,
    borderWidth: 2,
    paddingVertical: 10,
    paddingHorizontal: 20,
    margin: 10,
  },
  buttonText: {
    fontSize: 18,
    paddingHorizontal: 5,
    fontWeight: "bold",
    textAlign: "center",
  },
});

export default CustomButton;
