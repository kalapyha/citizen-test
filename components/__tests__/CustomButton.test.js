import React from "react";
import renderer from "react-test-renderer";
import CustomButton from "../CustomButton";

describe("<CustomButton />", () => {
  it("has 2 childs", () => {
    const tree = renderer
      .create(<CustomButton title="tt" onPress={() => null} />)
      .toJSON();
    expect(tree.children.length).toBe(2);
  });

  it("renders correctly", () => {
    const tree = renderer
      .create(<CustomButton title="test" onPress={() => null} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
