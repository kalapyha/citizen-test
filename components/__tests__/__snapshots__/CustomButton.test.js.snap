// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomButton /> renders correctly 1`] = `
<View
  accessibilityState={
    {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#2572b4",
      "borderColor": "#2572b4",
      "borderRadius": 5,
      "borderWidth": 2,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "margin": 10,
      "opacity": 1,
      "paddingHorizontal": 20,
      "paddingVertical": 10,
    }
  }
>
  <Text
    style={
      [
        {
          "fontSize": 18,
          "fontWeight": "bold",
          "paddingHorizontal": 5,
          "textAlign": "center",
        },
        {
          "color": "#fff",
        },
        {},
      ]
    }
  >
    test
  </Text>
  <View
    style={
      [
        {
          "marginLeft": 5,
        },
        {},
      ]
    }
  />
</View>
`;
